code/game/machinery/doors/firedoor.dm

```diff
@@ -479,18 +479,16 @@
 					"<span class='notice'>[user] finishes the firelock.</span>", \
 					"<span class='notice'>You finish the firelock.</span>")
 				playsound(get_turf(src), 'sound/items/deconstruct.ogg', 50, TRUE)
-				if(reinforced)
-					new /obj/machinery/door/firedoor/heavy(get_turf(src))
-				else
-					var/obj/machinery/door/firedoor/F = new firelock_type(get_turf(src))
+				var/obj/machinery/door/firedoor/F = new firelock_type(get_turf(src))
+				if(istype(F, /obj/machinery/door/firedoor/border_only))
 					F.dir = src.dir
 					F.update_icon()
 				qdel(src)
 				return
-			if(istype(C, /obj/item/stack/sheet/plasteel))
+			if(istype(C, /obj/item/stack/sheet/plasteel) && type != /obj/structure/firelock_frame/border)
 				var/obj/item/stack/sheet/plasteel/P = C
```

Comment:
I understand this isnt your code but since you are already touching this file can you cleanup all of the old style var names